extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -238,18 +243,44 @@ public boolean isRunning() {
             } catch (IllegalArgumentException e) {
                 throw new IllegalArgumentException("Cannot parse cron expression: " + cron, e);
             }
-            return Optional.of(new CronTrigger(id, start, cronExpr));
+            return Optional.of(new CronTrigger(id, start, cronExpr,
+                    SchedulerUtils.parseOverdueGracePeriod(scheduled, defaultGracePeriod)));
         } else if (!scheduled.every().isEmpty()) {
             final OptionalLong everyMillis = SchedulerUtils.parseEveryAsMillis(scheduled);
             if (!everyMillis.isPresent()) {
                 return Optional.empty();
             }
-            return Optional.of(new IntervalTrigger(id, start, everyMillis.getAsLong()));
+            return Optional.of(new IntervalTrigger(id, start, everyMillis.getAsLong(),
+                    SchedulerUtils.parseOverdueGracePeriod(scheduled, defaultGracePeriod)));
         } else {
             throw new IllegalArgumentException("Invalid schedule configuration: " + scheduled);
         }
     }
 
+    @Override
+    public List<Trigger> getScheduledJobs() {
+        List<Trigger> identities = new ArrayList<>();
+        for (ScheduledTask task : scheduledTasks) {
+            identities.add(task.trigger);
+        }
+        return identities;
```

Comment:
The result is not immutable. So something like `return List.copyOf(identities)` or  `return scheduledTasks.stream().map(s -> s.trigger).collect(Collectors.toUnmodifiableList())`?

Reply:
When I looked at the code base, I haven't seen any stream so I thought it was not in the coding standard !